StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+API.swift

```diff
@@ -616,8 +641,57 @@ extension PaymentSheet {
             } else {
                 fatalError("Either confirmHandler or confirmHandlerForServerSideConfirmation must be non-nil when using deferred intents")
             }
+        }
+    }
+
+    private static func confirmClientSide(for deferredContext: DeferredIntentContext, with clientSecret: String) {
```

Comment:
I'm trying to think how to make this whole flow easier to follow in the code.  I think one 'master' method that all the `.deferred` cases in `confirm` call would help.  The highlighted method here is almost that, except we don't use it in the saved PM case.

The pattern we have here where callbacks call other methods makes it hard to follow the code. I think we should take advantage of async/await to eliminate that.

This 'master' method would look roughly like
```
func handleDeferredIntentConfirmation(paymentMethod: STPPaymentMethod?, completionHandler: ..) {
  do {
    // Create PM if necessary
    let paymentMethod = paymentMethod ? try await createPaymentMethod(...)
    // Get Intent client secret from merchant
    let clientSecret = try await fetchIntentClientSecretFromMerchant(...)
    if isServerSideConfirmation {
       // Handle next actions
    } else {
       // Confirm payment
    }
  } catch {
    completionHandler(.failure(error))
  }
}
```

Reply:
This looks good! But I don't understand what `fetchIntentClientSecretFromMerchant` would be doing. We can't fetch it from the merchant, only wait for it via the two callbacks.

Reply:
It'd look something like
```
func fetchIntentClientSecretFromMerchant(...) async throws -> String {
  try await withCheckedThrowingContinuation { continuation in
    intentConfiguration.confirmCallback(..) { result in
     continuation.resume(with: result)
    }
  }
}
```

Reply:
Switched to async APIs!